source/MaterialXGenShader/HwShaderGenerator.h

```diff
@@ -16,14 +16,21 @@ class HwShaderGenerator : public ShaderGenerator
 
 public:
     /// Set the maximum number of light sources that can be active at once.
-    void setMaxActiveLightSources(size_t count) { _maxActiveLightSources = count; }
+    void setMaxActiveLightSources(size_t count) 
+    { 
+        _maxActiveLightSources = count; 
```

Comment:
A 0 value causes an code gen to try and create an array of size 0 which will fail compile.

Reply:
Nitpick: could write as `_maxActiveLightSources = std::max(1,count);`